src/Storage/StorageClient.php

```diff
@@ -217,4 +223,53 @@ public function createBucket($name, array $options = [])
         $response = $this->connection->insertBucket($options + ['name' => $name, 'project' => $this->projectId]);
         return new Bucket($this->connection, $name, $response);
     }
+
+    /**
+     * Register a StreamWrapper for reading and writing to Google Storage
+     *
+     * @param string $protocol The name of the protocol to use. Defaults to
+     *        'gs'.
+     * @throws \RuntimeException
+     */
+    public static function registerStreamWrapper(string $protocol = null)
+    {
+        $protocol = $protocol ?: 'gs';
+        if (!in_array($protocol, stream_get_wrappers())) {
+            if (!stream_wrapper_register($protocol, StreamWrapper::class)) {
+                throw new RuntimeException("Failed to register '$protocol://' protocol");
+            }
+        }
+    }
+
+    /**
+     * Unregisters the SteamWrapper
+     *
+     * @param string $protocol The name of the protocol to unregister. Defaults
+     *        to 'gs'.
+     */
+    public static function unregisterStreamWrapper(string $protocol = null)
+    {
+        stream_wrapper_unregister($protocol ?: 'gs');
+    }
+
+    /**
+     * Get the default client to use for global interactions with storage.
+     *
+     * @return StorageClient
+     */
+    public static function getDefaultClient()
+    {
+        return self::$defaultClient;
```

Comment:
I was thinking of a static method on the StreamWrapper, rather than the StorageClient. So when you call `registerStreamWrapper`, it works like this:

```php
// StorageClient.php
public function registerStreamWrapper() {
        $protocol = $protocol ?: 'gs';

        if (!in_array($protocol, stream_get_wrappers())) {

            StreamWrapper::setClient($this);
            if (!stream_wrapper_register($protocol, StreamWrapper::class)) {
                throw new RuntimeException("Failed to register '$protocol://' protocol");
            }
        }
}
```

Would something along those lines work?